packages/editor/src/components/post-author/test/index.js

```diff
@@ -41,14 +41,23 @@ describe( 'PostAuthor', () => {
 		},
 	};
 
+	const postAuthor = [
```

Comment:
This component has significantly increased in complexity, but its tests do not provide sufficient coverage.

Reply:
I am happy to work on additional tests - what do you think would be most useful to test? Thinking e2e tests that verify expected behavior with <50 users and over, I'll review what we have and suggest some additional tests.

Worth noting that the autocomplete component itself has 90% test coverage, so we probably don't need to test the component itself - only its integration into Gutenberg. https://github.com/alphagov/accessible-autocomplete

Reply:
A good target is a test case for every single possible combination of behaviors which could occur through the introduced functions / logic flows. Going in with this mindset can be a useful deterrent to complex interactions.

Reply:
Makes sense - I'll work on documenting what those are first

Reply:
I haven't had a chance to work on it, thanks for the reminder. Since the autoselect component itself is well tested, it seems like what we need to test here is that it is loaded correctly and that all the helpers work (get getInitialAuthor, suggestAuthor, getCurrentAuthor

Reply:
@aduth what were you thinking of in terms of test coverage? Component tests, or a few E2Es?

For E2Es I suppose that might look like:
- Select an author when < 50 users (mouse events)
- Select an author not included in the first response when > 50 users (mouse events)
- Select an author when < 50 users (keyboard events)
- Select an author not included in the first response when > 50 users (keyboard events)

Does this sound about right?

@adamsilverstein happy to add on a few of these, though would you prefer that as a commit to the branch directly, or a child PR?

Reply:
@gziolo I'm leaning more toward component tests here unless we have a clean example of E2E site setup for 5 users, then 500 on the same site.

Reply:
> I'm leaning more toward component tests here unless we have a clean example of E2E site setup for 5 users, then 500 on the same site.

I bet we only have one or a few users when running e2e tests. We can always add more of them using WP CLI during the setup of the Docker container to speed up things. Otherwise, you would have to write logic which switches to the admin account and is doing the dance to add more users. I'm not sure if this would be performant enough. 

Ideally, we have a mix of tests where component tests cover this edge cases you listed and e2e tests does some basic interactions testing.

_Aside:_ I don't think there should be 2 different components rendered depending on the number of results. @adamsilverstein is it the final proposal? I'm sure you had more PRs exploring alternative solutions.

Reply:
@gziolo Most E2E suites traditionally can be run on the same instance without needing to reset site state. 

This is why I'm leaning more toward component tests here as we would want to exercise UI pieces in the <50 and  >50 user scenario.

Reply:
> @adamsilverstein happy to add on a few of these, though would you prefer that as a commit to the branch directly, or a child PR?

Thank you for the offer of help :) Please feel free to make commits directly to the branch!

Reply:
> Aside: I don't think there should be 2 different components rendered depending on the number of results. @adamsilverstein is it the final proposal? > I'm sure you had more PRs exploring alternative solutions.

I have tried several approaches here. The reasoning behind two different components depending on the number of users is that when the number of options in a dropdown is small, regular HTMl selectors are an excellent choice, are semantically correct and properly accessible as they are browser native. Once the select grows beyond a certain size a auto-completing selector is much more user friendly (and still needs to remain fully accessible). 

In addition, using two separate components based on the number of items is the official recommendation from the accessible autocomplete repository, although they do say 'a few hundred' so my choice of 50 as the cutoff may be too low. I'd like to receive more feedback or even perform user testing to validate that. One WordPress specific consideration is that the REST API only returns 100 items max by default, so we may want the cutoff around that to avoid paginated requests on load. See https://github.com/alphagov/accessible-autocomplete#progressive-enhancement

Reply:
> One WordPress specific consideration is that the REST API only returns 100 items max by default, so we may want the cutoff around that to avoid paginated requests on load. See https://github.com/alphagov/accessible-autocomplete#progressive-enhancement

It might a better way to set the limit from the technical perspective. The only question is how likely it is that 100 items in the `select` can be considered easy to explore from the user's perspective. We can gather feedback and decide later though :)